evm/src/cpu/kernel/asm/core/call.asm

```diff
@@ -29,19 +29,22 @@ global sys_call:
     // stack: new_ctx, kexit_info, callgas, address, value, args_offset, args_size, ret_offset, ret_size
     // TODO: Consider call depth
 
+    %stack (new_ctx, kexit_info, callgas, address, value, args_offset, args_size, ret_offset, ret_size) ->
+          (new_ctx, args_offset, args_size, new_ctx, kexit_info, callgas, address, value, args_offset, args_size, ret_offset, ret_size)
+    %copy_mem_to_calldata
+    // stack: new_ctx, kexit_info, callgas, address, value, args_offset, args_size, ret_offset, ret_size
+    DUP5 DUP5 %address %transfer_eth %jumpi(panic) // TODO: Fix this panic.
+    DUP3 %set_new_ctx_gas_limit
+    %set_new_ctx_parent_pc(after_call_instruction)
+    %handle_precompiles
```

Comment:
Do you think there's a nice way we could call this with a smaller stack, ideally with only the data a precompile needs?

It feels a little error-prone if this same (kinda large) stack state must appear in each `sys_*call` and each precompile, and might be tricky if we wanted to change the stack layout.

I'm not sure if there's an elegant solution though; if not this is fine.

Reply:
Do you mean like DUPing `address, new_ctx, kexit_info, ret_offset, ret_size` (the info used by the precompiles) in front of the stack before calling `%handle_precompiles`? If so, yes that would work.

Reply:
Yeah, and I guess `handle_precompiles` could pop those if no precompile was executed. I don't think it'd save any code (might add a little), but feels a bit nicer to me if precompiles only receive the info they require. It's minor though, can keep it as is you prefer.

Reply:
Yeah I agree it's nicer. I've made the change 👍